### ISSUE TITLE
Developer

### ISSUE THREAD

**Issue by BrianCook** [dotnet/docs#5111]
This says nothing about the very inconvenient behavior which is that if you add a new setting, any saved values of your previous settings are all lost! You have to re-enter ALL the settings anytime you add a new one. Not friendly behavior, and not documented, nor documented what the workaround is.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: abb39304-acb2-1375-e604-656e718e07be
* Version Independent ID: ffcc8358-dc04-4caf-575c-219c8fcab699
* Content: [How To: Create a New Setting at Design Time](https://docs.microsoft.com/en-us/dotnet/framework/winforms/advanced/how-to-create-a-new-setting-at-design-time)
* Content Source: [docs/framework/winforms/advanced/how-to-create-a-new-setting-at-design-time.md](https://github.com/dotnet/docs/blob/master/docs/framework/winforms/advanced/how-to-create-a-new-setting-at-design-time.md)
* Service: **unspecified**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **dotnetcontent**

**Comment by mairaw** [comment]
@Tanya-Solyanik are you aware of any issues with that? 

@BrianCook can you provide more info on which version of VS you're using, language, etc.? I can't reproduce this with VS 2017. I can add new values without losing the old values.

**Comment by Tanya-Solyanik** [comment]
No, I'm not aware of this issue. I had experimented like this:
1 Added 2 settings to a settings file, saved the settings, restarted the application
2. read back the saved value
3. added a new setting value to the same setting file, rebuilt, restarted the app, the previously saved values are read back, and the new value is initialized to the value defined in the designer, as expected.
4. Added a new Settings1.Setting file, it did not affect the saved value. 

We definitely need more info here. @BrianCook - are you running your application as the same user before and after adding new settings values? Had the name your binary changed?

**Comment by BrianCook** [comment]
It is easy to duplicate in VS 2017, happens every time for me.

1. Add a string setting named "Setting1" to the Windows Forms program using
the settings designer. Leave default value as blank.
2. Run your program, save a new string value to Setting1.
3. Restart program, can read Setting1 with the new value.
4. Now add another string Setting2 to the settings, also with default value
blank.
5. Run your program, now if you read Setting1 it comes out blank with the
default value and you lost the settings you saved in step 2.

If I look in my User/MyName/AppData/Local folder under my app name, it
creates new subfolder with a different name every time I edit the settings
in the designer. Each new subfolder has the default settings and loses all
saved settings from before.

This is extremely annoying when those saved settings are long paths to
places in your file system and you have to re-pick them every time you add
a setting. I basically cut and paste between the settings files to save
time.

Thanks,

Brian

**Comment by Tanya-Solyanik** [comment]
I re-run your scenario and settings under user profile were not erased on steps 4-5. Are you incrementing assembly version of every re-build? How does the path to the stored settings look like? What version of the framework are you targeting?

> C:\Users\me\AppData\Local\WindowsFormsApp25\WindowsFormsApp25.exe_Url_yl2wkiizt1bkjvqiwmxjsaiggyrhktvh\1.0.0.0

**Comment by BillWagner** [comment]
@BrianCook Are you still having this issue? Did you try the suggestions from @Tanya-Solyanik ? If the issue is fixed, we'll close this. If not, we'd like to get to the root cause.

**Comment by BrianCook** [comment]
Yes, I still have the issue (although it does not bother me except when I
add settings)

I can try to put together a specific re-creation and send you a project
that exhibits the behavior, along with screenshots of my local settings
folder.

If I do this, what is the best way to send it to you? Do I need to create a
github project and share it with you?

Thanks,

Brian

**Comment by BillWagner** [comment]
Thanks @BrianCook 

Either a GitHub gist, or a GitHub project would be perfect. Thank you

**Comment by BillWagner** [comment]
@BrianCook Any updates?

**Comment by BrianCook** [comment]
I have attached a Word doc showing the problem in detail. Please let me
know if you need any other information.

Thanks,

Brian

**Comment by BillWagner** [comment]
@BrianCook Attachments  sent via mail aren't attached on the GitHub website. Can you visit this page and upload the doc?

Thanks.

**Comment by baddoggames** [comment]
I still think this is a real issue that should be addressed. Saving settings is a fundamental framework feature and the behavior is still badly designed and not documented.  Tagging @dotnet/docs per the email about the automatic purge #22351.

**Comment by BrianCook** [comment]
I left a comment on #5111 but it does not let me reopen the issue.

I still think this is a real issue that should be addressed. Saving
settings is a fundamental framework feature and the behavior is still badly
designed and not documented, as I carefully documented in the attached Word
doc. Why would anyone want or expect a settings feature to require you to
re-enter ALL your saved settings any time anything changes in your build?

Tagging @dotnet/docs per the email about the automatic purge #22351.
